### PR TITLE
IJPS-53 Functional tests for the DELETE /sittingRecords endpoint

### PR DESCRIPTION
**Pull request by marianadpereira**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -169,7 +169,7 @@ task functional(type: Test) {
         '--threads', '1',
         '--plugin', "json:${rootDir}/target/cucumber.json",
         '--plugin', "junit:${buildDir}/test-results/functional/cucumber.xml",
-        '--tags', '@Functional and not @Ignore',
+        '--tags', '@Functional1 and not @Ignore',
```
Comment (SomeshRA) [inline build.gradle:172]: Not sure whether this change is required

File: src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java [hunk 2]
```diff
@@ -86,6 +94,29 @@ public void recordForTheGivenHmctsServiceCodeExistsInTheDatabase() throws IOExce
             .then().log().all().assertThat().statusCode(201);
     }
 
+    @Given("a search is done on the hmctsServiceCode {string}, with the payload {string} to get the {string}")
+    public void searchIsDoneOnTheHmctsServiceCodeWithThePayloadToGetThe(String serviceCode, String payload, String
+        attribute) throws IOException {
+        String body = new
+            String(Files.readAllBytes(Paths.get("./src/functionalTest/resources/payloads/" + payload + ".json")));
+        body = body.replace("2023-03-10", randomDate);
```
Comment (SomeshRA) [inline src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java:102]: random date is not initialised

File: src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java [hunk 3]
```diff
@@ -107,12 +138,17 @@ public void theRequestContainsInvalidServiceToken() {
...
         given = request.pathParam(pathParam,value);
     }
 
     @When("the request body contains the {string} as in {string}")
     public void theRequestBodyContainsThe(String description, String fileName) throws IOException {
-        String body = new String(Files.readAllBytes(Paths.get("./src/functionalTest/resources/payloads/" + fileName)));
+        String body = new String(Files.readAllBytes(Paths.get("./src/functionalTest/resources/payloads/" + fileName
+                                                                  + ".json")));
         if (description.equalsIgnoreCase("payload matching data from existing record")) {
             body = body.replace("2023-03-10", randomDate);
```
Comment (SomeshRA) [inline src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java:153]: random date is not initialised

File: src/functionalTest/resources/features/F-003 - Delete sittingRecords endpoint/F-003.feature [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    And a record for the given hmctsServiceCode exists in the database with the payload "F-003_CreateRecord"
+    And a search is done on the hmctsServiceCode "ABA5", with the payload "F-005_allFields" to get the "sittingRecords[0].sittingRecordId"
+    When a request is prepared with appropriate values
+    And the request contains a valid service token
+    And the request contains the "sittingRecordId" as "id of the previously created record"
+    And a call is submitted to the "SittingRecord" endpoint using a "DELETE" request
+    Then a "positive" response is received with a "200 OK" status code
+
+  @S-003.3 @Ignore #AC03 Ignore until IJPS-62 is ready
+  Scenario: Success response - Return 200 success with Sitting Record update status as deleted for jps-admin
```
Comment (SomeshRA) [inline src/functionalTest/resources/features/F-003 - Delete sittingRecords endpoint/F-003.feature:27]: Shouldn't record status be 'SUBMITTED'

File: src/functionalTest/resources/test-config.properties [hunk 5]
```diff
@@ -19,5 +19,8 @@ idam.publisher.password=${JPS_PUBLISHER_PASSWORD:password}
 idam.admin.username=${JPS_RECORDER_USERNAME:jps-admin-role@gmail.com}
```
Comment (SomeshRA) [inline src/functionalTest/resources/test-config.properties]: I suppose this should be JPS_ADMIN_USERNAME

File: src/functionalTest/resources/test-config.properties [hunk 6]
```diff
@@ -19,5 +19,8 @@ idam.publisher.password=${JPS_PUBLISHER_PASSWORD:password}
...
 idam.admin.password=${JPS_RECORDER_PASSWORD:password}
```
Comment (SomeshRA) [inline src/functionalTest/resources/test-config.properties]: suppose this should be JPS_ADMIN_PASSWORD

File: src/functionalTest/resources/test-config.properties [hunk 7]
```diff
@@ -19,5 +19,8 @@ idam.publisher.password=${JPS_PUBLISHER_PASSWORD:password}
...
 
+idam.johAdmin.username=${JPS_RECORDER_USERNAME:jps-JOH-admin-role@gmail.com}
```
Comment (SomeshRA) [inline src/functionalTest/resources/test-config.properties]: I suppose this should be JPS_JOH_ADMIN_USERNAME

File: src/functionalTest/resources/test-config.properties [hunk 8]
```diff
@@ -19,5 +19,8 @@ idam.publisher.password=${JPS_PUBLISHER_PASSWORD:password}
...
+idam.johAdmin.password=${JPS_RECORDER_PASSWORD:password}
```
Comment (SomeshRA) [inline src/functionalTest/resources/test-config.properties]: I suppose this should be JPS_JOH_ADMIN_PASSWORD

File: src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java (hunk 3 above)
Comment (marianadpereira) [reply src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java:153]: Same as above.

File: src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java (hunk 2 above)
Comment (SomeshRA) [reply src/functionalTest/java/uk/gov/hmcts/reform/jps/StepDefinitions.java:102]: Not sure as I am not able to see where its initialised

**Review by SomeshRA — APPROVED** [review]
LGTM

**Pull request merged**